File: src/runtime/php/std/os_native.php
<?php
declare(strict_types=1);
/**
 * Native implementation of pytra.std.os for PHP.
 *
 * source: src/pytra/std/os.py
 */

if (!function_exists('__native_getcwd')) {
    function __native_getcwd(): string {
        return (string)getcwd();
    }
}

if (!function_exists('__native_mkdir')) {
    function __native_mkdir($p): void {
        $path = (string)$p;
        if (!is_dir($path)) {
            mkdir($path);
        }
    }
}

if (!function_exists('__native_makedirs')) {
    function __native_makedirs($p, $exist_ok = false): void {
        $path = (string)$p;
        if ($exist_ok && is_dir($path)) {
            return;
        }
        mkdir($path, 0777, true);
    }
}

File: src/runtime/go/std/os_native.go
// os_native.go: @extern delegation for pytra.std.os.
// Hand-written native implementation.
package main

import "os"

func getcwd() string {
	dir, err := os.Getwd()
	if err != nil {
		panic(err)
	}
	return dir
}
func py_getcwd() string { return getcwd() }

func mkdir(path string, exist_ok bool) {
	if exist_ok {
		if _, err := os.Stat(path); err == nil {
			return
		}
	}
	if err := os.Mkdir(path, 0755); err != nil {
		panic(err)
	}
}
func py_mkdir(path string, exist_ok bool) { mkdir(path, exist_ok) }

func makedirs(path string, exist_ok bool) {
	_ = exist_ok
	if err := os.MkdirAll(path, 0755); err != nil {
		panic(err)
	}
}
func py_makedirs(path string, exist_ok bool) { makedirs(path, exist_ok) }
